### PR TITLE
Add warning for large benchmark families

### PR DESCRIPTION
**Pull request by EricWF**
`->Ranges(...)` generates a list of arguments which is the Cartesian product of  all of the supplied ranges. 
If the list contains 100 items then the benchmark will be run 100 times, which can be very slow and possibly unexpected. This patch adds a warning message when the generated list is "large".

Additionally this patch re-works the error output to use the reporters error stream instead of `std::cerr`.

### CODE REVIEW

File: src/benchmark.cc [hunk 1]
```diff
@@ -446,12 +452,23 @@ bool BenchmarkFamilies::FindBenchmarks(
...
-
-    for (auto const& args : family->args_) {
-      const std::vector<int>* thread_counts =
+    const std::vector<int>* thread_counts =
         (family->thread_counts_.empty()
          ? &one_thread
-         : &family->thread_counts_);
+         : &static_cast<const std::vector<int>&>(family->thread_counts_));
+    const size_t family_size = family->args_.size() * thread_counts->size();
+    // The benchmark will be run at least 'total' different inputs. If 'total'
```
Comment (dominichamon) [inline src/benchmark.cc]: change total to family_size in the comment

File: src/benchmark.cc [hunk 2]
```diff
@@ -1093,47 +1110,52 @@ size_t RunSpecifiedBenchmarks(BenchmarkReporter* console_reporter,
...
-  } else {
-    // Setup the reporters
-    std::ofstream output_file;
-    std::unique_ptr<BenchmarkReporter> default_console_reporter;
-    std::unique_ptr<BenchmarkReporter> default_file_reporter;
-    if (!console_reporter) {
-      auto output_opts = FLAGS_color_print ? ConsoleReporter::OO_Color
-                                           : ConsoleReporter::OO_None;
-      default_console_reporter = internal::CreateReporter(
+  // Setup the reporters
```
Comment (dominichamon) [inline src/benchmark.cc]: we do a lot of work now just to list the benchmarks.. why?

Comment (EricWF) [inline src/benchmark.cc]: I was to replace usage of `std::cout`/`std::cerr` with the reporters output/error streams, even when just listing the tests.

**Comment by dominichamon** [comment]
LGTM

**Pull request merged**